packages/esm-service-queues-app/src/patient-search/visit-form-queue-fields/visit-form-queue-fields.component.tsx

```diff
@@ -1,47 +1,54 @@
-import React, { useEffect, useState } from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import classNames from 'classnames';
 import { InlineNotification, Select, SelectItem, RadioButtonGroup, RadioButton, TextInput } from '@carbon/react';
 import { useQueueLocations } from '../hooks/useQueueLocations';
 import styles from './visit-form-queue-fields.scss';
-import { useConfig, useLayoutType, ResponsiveWrapper } from '@openmrs/esm-framework';
+import { useConfig, ResponsiveWrapper, useSession } from '@openmrs/esm-framework';
 import { useTranslation } from 'react-i18next';
 import { useQueues } from '../../hooks/useQueues';
 import { type ConfigObject } from '../../config-schema';
 import { SelectSkeleton } from '@carbon/react';
 import { RadioButtonSkeleton } from '@carbon/react';
+import { AddPatientToQueueContext } from '../patient-search.workspace';
 
-const StartVisitQueueFields: React.FC = () => {
+const VisitFormQueueFields: React.FC = () => {
   const { t } = useTranslation();
-  const isTablet = useLayoutType() === 'tablet';
   const { queueLocations, isLoading: isLoadingQueueLocations } = useQueueLocations();
+  const { sessionLocation } = useSession();
   const config = useConfig<ConfigObject>();
-  const defaultStatus = config.concepts.defaultStatusConceptUuid;
-  const defaultPriority = config.concepts.defaultPriorityConceptUuid;
-  const emergencyPriorityConceptUuid = config.concepts.emergencyPriorityConceptUuid;
   const [selectedQueueLocation, setSelectedQueueLocation] = useState(queueLocations[0]?.id);
   const { queues, isLoading: isLoadingQueues } = useQueues(selectedQueueLocation);
-  const [priority, setPriority] = useState(defaultPriority);
-  const [status, setStatus] = useState(defaultStatus);
+  const [priority, setPriority] = useState(config.concepts.defaultPriorityConceptUuid);
+  const [status, setStatus] = useState(config.concepts.defaultStatusConceptUuid);
   const [sortWeight, setSortWeight] = useState(0);
-  const [service, setSelectedService] = useState('');
-  const priorities = queues.find((q) => q.uuid === service)?.allowedPriorities ?? [];
-  const statuses = queues.find((q) => q.uuid === service)?.allowedStatuses ?? [];
+  const [selectedService, setSelectedService] = useState('');
+  const priorities = queues.find((q) => q.uuid === selectedService)?.allowedPriorities ?? [];
+  const statuses = queues.find((q) => q.uuid === selectedService)?.allowedStatuses ?? [];
+  const { currentServiceQueueUuid } = useContext(AddPatientToQueueContext);
 
   useEffect(() => {
-    if (priority === emergencyPriorityConceptUuid) {
+    if (priority === config.concepts.emergencyPriorityConceptUuid) {
       setSortWeight(1);
     }
   }, [priority]);
 
   useEffect(() => {
     if (queues?.length > 0) {
-      setSelectedService(queues[0].uuid);
```

Comment:
I'm not entirely sure this solves the issue presented in the ticket @brandones .  The issue is that the active visits table is not limited to a specific queue, but allows the user to add a new queue entry without specifying which queue they want to add the queue entry into.  As far as I can tell, that behavior remains after these changes, as below the `selectedService` just defaults to the first queue available in the given location, and there is nothing in place that prompts the user to select the queue / service of interest in this situation.

Reply:
Oh, I think I understand what you meant in the ticket now. At present the behavior is that when a patient is added from a view that has multiple queues, there is a default value for the queue to add to, which happens to be the first queue in the list. The desired behavior is that there should be no default value for the queue when the user is in a context with multiple queues.